public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -150,17 +153,57 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
   }
 
   query(request: DataQueryRequest<InfluxQuery>): Observable<DataQueryResponse> {
+    // for not-flux queries we call `this.classicQuery`, and that
+    // handles the is-hidden situation.
+    // for the flux-case, we do the filtering here
+    const filteredRequest = {
+      ...request,
+      targets: request.targets.filter((t) => t.hide !== true),
+    };
+
     if (this.isFlux) {
-      // for not-flux queries we call `this.classicQuery`, and that
-      // handles the is-hidden situation.
-      // for the flux-case, we do the filtering here
-      const filteredRequest = {
-        ...request,
-        targets: request.targets.filter((t) => t.hide !== true),
-      };
       return super.query(filteredRequest);
     }
 
+    if (config.featureToggles.influxBackendMigration && this.access === 'proxy' && request.app === CoreApp.Explore) {
+      return super.query(filteredRequest).pipe(
+        map((res) => {
+          if (res.error) {
+            throw {
+              message: 'InfluxDB Error: ' + res.error.message,
+              res,
+            };
+          }
+
+          const seriesList: any[] = [];
+
+          if (res.data && res.data.length > 0) {
+            filteredRequest.targets.forEach((target) => {
+              const filteredFrames = res.data.filter((t) => t.refId === target.refId);
```

Comment:
we could probably do this a little faster by grouping the items in `res.data` by the `refId` before the `forEach`, something like:
```javascript
import {groupBy} from 'lodash'


const groupedFrames = groupBy(res.data, x => x.refId);
filteredRequest.targets.forEach((target) => {
    const filteredFrames = groupedFrames[target.refId] ?? [];
   switch...
});]


...but maybe this is too much optimalization, considering usually there will be only a single `refId`. pick any option you prefer 😄

Reply:
I definitely agree. I'm in favour of making this faster, even if it's only a little bit :)